libraries/eosiolib/contracts/eosio/key_value.hpp

```diff
@@ -85,16 +81,23 @@ namespace detail {
 /**
  * The key_type struct is used to store the binary representation of a key.
  */
-struct key_type : private std::string {
+struct key_type : private std::vector<char> {
    key_type() = default;
-   key_type(const char* c, size_t s) : std::string(c, s) {}
+
+   key_type(std::vector<char>&& v) : std::vector<char>(v) {}
 
    key_type operator+(const key_type& b) const {
       key_type ret = *this;
       ret += b;
       return ret;
    }
 
+   key_type& operator+=(const key_type& b) const {
+      this->resize(this->size()+b->size());
```

Comment:
Did you mean reserve?  You can just remove this.  reserve isn't really useful here.